pytest_fxa/plugin.py

```diff
@@ -69,23 +78,47 @@ def fxa_email(pytestconfig):
 
 @pytest.fixture
 def fxa_account(fxa_client, fxa_email):
+    verified = False
+    timeout = time.time()
     logger = logging.getLogger()
-    account = TestEmailAccount(email=fxa_email)
-    password = ''.join([random.choice(string.ascii_letters) for i in range(8)])
-    FxAccount = collections.namedtuple('FxAccount', 'email password')
-    fxa_account = FxAccount(email=account.email, password=password)
-    session = fxa_client.create_account(fxa_account.email,
-                                        fxa_account.password)
-    logger.info('Created: {}'.format(fxa_account))
-    account.fetch()
-    message = account.wait_for_email(lambda m: 'x-verify-code' in m['headers'])
-    session.verify_email_code(message['headers']['x-verify-code'])
+    while verified is False:
+        if timeout + 30 == time.time():
+            # Verification keeps failing, we should clear the restmail address
+            # because it probably didn't get deleted
```

Comment:
So my thoughts for this were that if the verification fails, we have to delete and try again. The PyFxA [function](https://github.com/mozilla/PyFxA/blob/b2fda137eeaf1a3e428a27a68508df5bd9fcb2f7/fxa/tests/utils.py#L99) doesn't look like it goes through different accounts if more than one is created. That is also why I have the ```requests.delete```.

Reply:
If everything works it runs at about the same time as before

Reply:
I still don't understand why we're generating a password/account when the verifcation fails. Couldn't we just retry the verification part?

Reply:
I'd perhaps be more comfortable with this patch focusing on the cleanup, and a separate patch for adding a retry feature. We do have issues occaisionally, which a retry may help with, but I wonder if we should have a backoff factor.

Reply:
We could retry the verification but the function I linked already does that. And I don't think retrying it would let us retry a different email's response headers. There isn't a lot of error catching with pyfxa tbh

Reply:
If the PyFxA method already retries verification then I don't understand why we're adding a retry ourselves. I think the fix here should just be to ensure we cleanup when things don't work, and provide useful error messages.

Reply:
From what I was told it just waits for the first email, it doesn't go through a list or anything.

Reply:
For my application my tests are not run in parallel and I need to use the same email. Since these methods and such are function scoped I'd have to copy them verbatim just to scope them as ```session```.

Reply:
It's Saturday folks! Enjoy your weekend! 🏕

(I know you're traveling @davehunt 🚆)

Reply:
Heh, I was about to say the same @hackebrot :smile

Reply:
I'm fine with the email address being static so long as the risks and consequences are understood. I still don't see why this retry loop is necessary though if we perform the necessary cleanup every time regardless of the outcome.

Reply:
Well if the verify fails, even if we do correct cleanup, what would be returned to the user? I have the loop to actually get the user an email they can use.